### PR TITLE
Issue 73 (Multiple type of event counts)

### PR DESCRIPTION
**Pull request by Rolletta**

### CODE REVIEW

File: R/simulate_test_data.R [hunk 1]
```diff
@@ -668,8 +701,8 @@ sim_test_data_portfolio <- function(df_config, df_ae_rates = NULL, parallel = FA
               n_sites = 1,
               max_visit_mean = max_visit_mean,
               max_visit_sd = max_visit_sd,
-              ae_per_visit_mean = ae_per_visit_mean,
```
Comment (erblast) [inline R/simulate_test_data.R:671]: argument names for existing functions must not be changed

File: tests/testthat/test_S3_simaerep.R [hunk 2]
```diff
@@ -104,18 +104,18 @@ test_that("plot.simaerep throws error when original visit data cannot be retriev
 test_that("simaerep() with mult_corr = FALSE must not return adjusted probabilities", {
 
   aerep <- simaerep(df_visit_test, mult_corr = FALSE)
-  expect_true(! "prob_low_adj" %in% colnames(aerep$df_eval))
```
Comment (erblast) [inline tests/testthat/test_S3_simaerep.R:107]: the column names generated should still be the sames

File: R/S3_simaerep.R [hunk 3]
```diff
@@ -128,29 +135,45 @@ validate_simaerep <- function(x) {
...
+                      param_eval_sites = list(
+                        method = "BH"
+                      ),
+                      env = parent.frame(),
+                      event_names = c("ae")
+) {
 
   call <- rlang::enexpr(df_visit)
 
+  if (ncol(df_visit) != 7 + 2 * length(event_names)) {
```
Comment (erblast) [inline R/S3_simaerep.R:163]: df_visit should be allowed to have extra columns, more than 7, please remove check

File: R/S3_simaerep.R [hunk 4]
```diff
@@ -128,29 +135,45 @@ validate_simaerep <- function(x) {
...
+    stop(paste0("Different number of event names (", length(event_names),
+                ") than expected (", (ncol(df_visit) - 7) / 2, ")"))
+  }
+  for (x in event_names) {
```
Comment (erblast) [inline R/S3_simaerep.R:167]: the same check is repeated in check_df_visit, remove

File: R/simaerep.R [hunk 5]
```diff
@@ -388,47 +393,75 @@ get_visit_med75 <- function(df_pat,
 eval_sites <- function(df_sim_sites,
                         method = "BH",
                         under_only = TRUE,
+                        event_names = c("ae"),
                         ...) {
 
-  df_out <- df_sim_sites
+event_names_mod <- character()
```
Comment (erblast) [inline R/simaerep.R:399]: fix indentation rows 399-413

File: R/simaerep.R [hunk 6]
```diff
@@ -499,18 +536,23 @@ warning_na <- function(df, col) {
...
+        across(.cols = all_of(col2), .fns = ~sum(.x, na.rm = TRUE), .names = "{col2}")
       ) %>%
       mutate(
-        any_na = .data$any_na > 0
-      ) %>%
-      pull(.data$any_na)
+        across(.cols = all_of(col2), .fns = ~(.x > 0), .names = "{col2}"))
+
+
+if (length(col) > 1) return(any(any_na[col2]))
```
Comment (erblast) [inline R/simaerep.R:554]: fix indentation

File: R/simaerep.R [hunk 7]
```diff
@@ -388,47 +393,75 @@ get_visit_med75 <- function(df_pat,
 eval_sites <- function(df_sim_sites,
                         method = "BH",
                         under_only = TRUE,
+                        event_names = c("ae"),
                         ...) {
 
-  df_out <- df_sim_sites
+event_names_mod <- character()
+colname_pval <- character()
```
Comment (erblast) [inline R/simaerep.R:400]: I like how you handled the problem here, I would just like you to change the naming of your variables here to use the plural. e.g. colnames_low

File: R/simaerep.R [hunk 8]
```diff
@@ -499,18 +536,23 @@ warning_na <- function(df, col) {
 
 get_any_na <- function(df, col) {
```
Comment (erblast) [inline R/simaerep.R:537]: you are basically vectorised this function which is fine, col was originally meant to be a character of length one. Please make this expressive and change col --> cols and do not use col2 (better would be cols_any_na).

Comment (erblast) [reply R/simaerep.R:537]: this function is also used by check_df_visit(), here purrr::map is used instead, this is now not necessary anymore, please adapt the code here

**Comment by erblast** [comment]
Hi Adam,

I finished my review. I have opened a PR with unit tests to your repo.

**1. Final Column Names**
I would expect if I pass more than one event name that the name ("ae") is always added to the column name. If there is only one event name, the event does not need to be added to the column name (default).

**2. Multiple event_names and inframe == FALSE**
I think its better to revert back to the original here. So event_names should only be passed to the inframe method relevant functions: check_df_visit, site_aggr, eval_sites.

**3. orivisit S3 class**
This class tries to find the original data from the calling environment for plotting. This currently does not work for event_names with length > 1. I think you will need to add event_names as an attribute to the S3 class. Make sure event_names is always passed to summarise_df_visit()

**Comment by Rolletta** [comment]
Hi @erblast, 
I have added a new parameter for the plotting functions that specifies which event/s should be plotted

File: R/simaerep.R [hunk 9]
```diff
@@ -33,23 +33,23 @@ if (getRversion() >= "2.15.1") {
 #'
 #' @rdname check_df_visit
 #' @export
-check_df_visit <- function(df_visit) {
+check_df_visit <- function(df_visit, event_names = "ae") {
 
   df_visit <- ungroup(df_visit)
-
+  colname <- paste0("n_", event_names)
```
Comment (erblast) [inline R/simaerep.R:39]: change colname to colnames, to show it can be multiple

File: R/simaerep.R [hunk 10]
```diff
@@ -360,12 +366,12 @@ get_visit_med75 <- function(df_pat,
...
-#'   \item{**pval_adj**}{adjusted p-values}
-#'   \item{**prob_low_adj**}{adjusted bootstrapped probability for having mean_ae_site_med75 or lower}
-#'   \item{**pval_prob_ur**}{probability under-reporting as 1 - pval_adj, poisson.test (use as benchmark)}
-#'   \item{**prob_low_prob_ur**}{probability under-reporting as 1 - prob_low_adj, bootstrapped (use)}
+#'   \item{**ae_pval**}{p-value as returned by \code{\link[stats]{poisson.test}}}
+#'   \item{**ae_prob_low**}{bootstrapped probability for having mean_ae_site_med75 or lower}
+#'   \item{**ae_pval_adj**}{adjusted p-values}
+#'   \item{**ae_prob_low_adj**}{adjusted bootstrapped probability for having mean_ae_site_med75 or lower}
+#'   \item{**ae_pval_prob_ur**}{probability under-reporting as 1 - pval_adj, poisson.test (use as benchmark)}
+#'   \item{**ae_prob_low_prob_ur**}{probability under-reporting as 1 - prob_low_adj, bootstrapped (use)}
```
Comment (erblast) [inline R/simaerep.R:374]: The ae_ prefix can be removed

File: tests/testthat/test_event_names.R [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+    under_only = FALSE,
+    inframe = TRUE,
+    mult_corr = TRUE
+  )
+
+  expect_s3_class(plot(aerep, what = "ur", study = "A", event_names = events, plot_event = "ae"), "ggplot")
+  expect_s3_class(plot(aerep, what = "ur", study = "A", event_names = events, plot_event = "y"), "ggplot")
+  expect_s3_class(plot(aerep, what = "ur", study = "A", event_names = events), "ggplot")
+
+  expect_s3_class(plot(aerep, what = "med75", study = "A", event_names = events, plot_event = events), "ggplot")
```
Comment (erblast) [inline tests/testthat/test_event_names.R:181]: med75 should be the same regardless of event, therefore the plot should work w/o passing event_names and plot_event

File: tests/testthat/test_event_names.R [hunk 12]
```diff
@@ -0,0 +1,235 @@
...
+
+})
+
+
+
+test_that("event_names works with duckdb backend", {
```
Comment (erblast) [inline tests/testthat/test_event_names.R:187]: is this test on your computer as slow as it is for me? It takes a couple of minutes to execute? I think the last statement with visit_med75 slows it down a lot. The result looks fine though. Nevertheless if its as slow for you as it is for me I need to investigate this. This is an edge case that is allowed to be slow for all I car just need to check the exisiting duckdb steps to check whether anything else slowed down as well.

File: tests/testthat/test_event_names.R (hunk 11 above)
Comment (Rolletta) [reply tests/testthat/test_event_names.R:181]: Hi @erblast ,  doesn't this plot have cumulative AEs on the Y axis? So the final plots would be different depending on the plot_event parameter

Comment (erblast) [reply tests/testthat/test_event_names.R:181]: Yes you are right, plot_event is needed event_names not

**Comment by erblast** [comment]
https://github.com/openpharma/simaerep/issues/75

**Comment by erblast** [comment]
https://github.com/openpharma/simaerep/issues/75

**Comment by erblast** [comment]
https://github.com/openpharma/simaerep/issues/73
